planetiler-core/src/main/java/com/onthegomap/planetiler/archive/TileArchiveWriter.java

```diff
@@ -303,8 +301,9 @@ private void tileWriter(Iterable<TileBatch> tileBatches) throws ExecutionExcepti
         TileEncodingResult encodedTile;
         while ((encodedTile = encodedTiles.poll()) != null) {
           TileCoord tileCoord = encodedTile.coord();
-          assert lastTile == null || lastTile.compareTo(tileCoord) < 0 : "Tiles out of order %s before %s"
-            .formatted(lastTile, tileCoord);
+          assert lastTile == null ||
+            order.encode(tileCoord) > order.encode(lastTile) : "Tiles out of order %s before %s"
```

Comment:
Technically the PMTiles writer can take out-of-order writes and has some logic to set `clustered=false` and re-sort the entries. If we are enforcing this via assertion then we could take that logic out.